scripts/migration/dataset_paths_ecs_rollback.py

```diff
@@ -0,0 +1,259 @@
+
+    print("Found: {} {} documents for a potential path change. In progress ... ".format(docs_count, entity_name))
+
+    patched = 0
+    failed_count = 0
+    for item in docs:
+        # item preview
+        if verbose:
+            print("Rollbacking path changes for {} '{}' v{} (_id={}).".format(entity_name, item["name"], item["version"], item["_id"]))
+
+        update_data_set, update_data_unset = update_data_for_item(item, mapping_settings)
+        if update_data_set != {} and update_data_unset != {}:
+            if dryrun:
+                print("  *would rollback* {}".format(data_update_to_nice_string(update_data_set)))
+                print("")
+
+            else:
+                try:
+                    if verbose:
+                        print("  *rollbacking*: {}".format(data_update_to_nice_string(update_data_set)))
+
+                    update_result = dataset_collection.update_one(
+                        {"$and": [
+                            {"_id": item["_id"]}
+                        ]},
+                        {"$set": update_data_set, "$unset": update_data_unset}
+                    )
+                    if update_result.acknowledged and verbose:
+                        print("Successfully rollbacked changed path for {} '{}' v{} (_id={}).".format(entity_name, item["name"], item["version"], item["_id"]))
+                        print("")
+
+                except Exception as e:
+                    print("Warning: Error while change-path rollbacking for {} '{}' v{} (_id={}): {}".format(entity_name, item["name"], item["version"], item["_id"], e))
+                    failed_count += 1
+                else:
+                    patched += 1
+        else:
+            if verbose:
+                print("Nothing left to rollback for {} '{}' v{} (_id={}).".format(entity_name, item["name"], item["version"], item["_id"]))
+
+    print("Successfully rollbacked {} of {} {} entries, failed: {}".format(patched, docs_count, entity_name, failed_count))
+    print("")
+
+
+def rollback_pathchange_collections_by_ds_names(target_db: MenasDb,
+                                                supplied_ds_names: List[str],
+                                                mapping_settings: RollbackSettings,
+                                                onlydatasets: bool,
+                                                dryrun: bool) -> None:
+
+    if verbose:
+        print("Dataset names given: {}".format(supplied_ds_names))
+
+    ds_names_found = target_db.get_distinct_ds_names_from_ds_names(supplied_ds_names, migration_free_only=False)
+    print('Dataset names to rollback path-changes (actually found db): {}'.format(ds_names_found))
+
+    if onlydatasets:
+        print("Configured *NOT* to rollback path-changes for related mapping tables.")
+    else:
+        mapping_table_found_for_dss = target_db.get_distinct_mapping_tables_from_ds_names(ds_names_found, migration_free_only=False)
```

Comment:
Here I was not able to get some MT as an output.
```
python3 ./dataset_paths_ecs_rollback.py mongodb://localhost:27017/admin -n -v -d CPS_castingConformanceRule coalesceConformanceRule roomsMappingTable -t menas -s Test another
```

where `roomsMappingTable` is Name of one of Mapping Table records.

I would like to check here. I were debugging it and I was not able to get data from `get_distinct_mapping_tables_from_ds_names - s_collection.aggregate call`, the ds_name filter not worked for me.

Reply:
Correct observation. This was fixed in https://github.com/AbsaOSS/enceladus/pull/2201/commits/05c2a4e77226b40f10a733315a964d64317008dd
by introducing `-m` to actually path-change MTs, it did not work correctly with `-d` as assumed.